scipy/special/specfun2.cpp

```diff
@@ -0,0 +1,118 @@
+#include <array>
+#include <map>
+#include <tuple>
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
```

Comment:
This is what's leading to the warning.

```
../scipy/special/specfun2.cpp:8: error: "NPY_NO_DEPRECATED_API" redefined [-Werror]
    8 | #define NPY_NO_DEPRECATED_API NPY_API_VERSION
```

A relevant issue https://github.com/scipy/scipy/pull/19810